test/index.js

```diff
@@ -170,4 +171,117 @@ describe('stylelint-webpack-plugin', function () {
         expect(stats.compilation.errors).to.have.length(0);
       });
   });
+
+  context('lintDirtyModulesOnly flag is enabled', function () {
+    it('skips linting on initial run', function () {
+      var config = {
+        context: './test/fixtures/test3',
+        entry: './index',
+        plugins: [
+          new StyleLintPlugin({
+            configFile: configFilePath,
+            quiet: true,
+            lintDirtyModulesOnly: true
+          }),
+          new webpack.NoErrorsPlugin()
+        ]
+      };
+
+      return pack(assign({}, baseConfig, config))
+        .then(function (stats) {
+          expect(stats.compilation.errors).to.have.length(0);
+        });
+    });
+
+    /*
+     @TODO: come back later to integration tests
+     @see https://github.com/vieron/stylelint-webpack-plugin/pull/53#discussion_r95084529
+     Dependencies to install or include for following test:
+          - `node-fs-extra`
+          - `../helpers/watch`
+          - `path`
+     // Webpack integration test.
+     it('lints only changed files in watch mode', function (done) {
```

Comment:
This could be marked with `it.skip` so we don't have to commit commented out code. In fact it could be inside it's own `context` block.

In that block we should _stub_ out the parts of webpack that we can assume will operate as a black box.

Then, I believe the `watchCallback` can be [spied on and assertions can be made on individual calls of that fn](http://devdocs.io/sinon/index#sinon-spy). I don't mean to prescribe `sinon` as a solution since other libraries exist (e.g. I prefer [testdouble](https://github.com/testdouble/testdouble.js/)) but I just wanted to demonstrate that I think it's possible to write this test without all the imperative code and then we can merge it :)